mantidproject/mantid
# Correct updating issues in muon analysis with current dataset index and result tab function

**Based of #28968 so merge after that**
**Description of work.**
This corrects two minor issues with the Muon Analysis and Frequency domain analysis browsers. See commit messages for further details. 



**To test:**
* See the issue #29015 for details on how to produce the first bug.
* To test the second issue load MUSR62260 into muon analysis
* Go to the fitting tab and change the workspace you are currently displaying, then add a function.
* Plot guess and check it updates as you change the function parameters.



Fixes #29015  


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
